crates/wasm-mutate/src/lib.rs

```diff
@@ -287,19 +282,65 @@ impl<'wasm> WasmMutate<'wasm> {
                 RemoveItemMutator(Item::Data),
                 RemoveItemMutator(Item::Element),
                 RemoveItemMutator(Item::Tag),
+                ModifyDataMutator,
             )
         );
 
         Err(Error::no_mutations_applicable())
     }
 
+    fn setup(&mut self, input_wasm: &'wasm [u8]) -> Result<()> {
+        self.info = Some(ModuleInfo::new(input_wasm)?);
+        self.rng = Some(SmallRng::seed_from_u64(self.seed));
+        Ok(())
+    }
+
     pub(crate) fn rng(&mut self) -> &mut SmallRng {
         self.rng.as_mut().unwrap()
     }
 
     pub(crate) fn info(&self) -> &ModuleInfo<'wasm> {
         self.info.as_ref().unwrap()
     }
+
+    fn raw_mutate(&mut self, data: &mut Vec<u8>) -> Result<()> {
+        // If a raw mutation function is configured then that's prioritized.
+        if let Some(mutate) = &self.raw_mutate_func {
+            return mutate(data);
+        }
+
+        // If no raw mutation function is configured then we apply a naive
+        // default heuristic. For now that heuristic is to simply replace a
+        // subslice of data with a random slice of other data.
+        //
+        // First up start/end indices are picked.
+        let start = self.rng().gen_range(0, data.len() + 1);
+        let end = if start == data.len() {
+            data.len()
+        } else {
+            self.rng().gen_range(start, data.len() + 1)
+        };
```

Comment:
What do you think of this?

```suggestion
        let a = self.rng().gen_range(0, data.len() + 1);
        let b = self.rng().gen_range(0, data.len() + 1);
        let start = a.min(b);
        let end = a.max(b);
```